### PR TITLE
feat: js-ipfs support of CIDs in /ipns/ content paths

### PR DESCRIPTION
**Pull request by lidel**
### Motivation

We want root identifiers of content paths to be usable in case-insensitive contexts such as [subdomains](https://github.com/ipfs/in-web-browsers/issues/89). 
While we can use [CIDv1 encoded in Base32](https://github.com/ipfs/ipfs/issues/337) in `/ipfs/` namespace, Identifiers in `/ipns/` are still hardcoded to Base58btc,

Libp2p approved [RFC 0001: Text Peer Ids as CIDs](https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md) recently, and we need to implement it in js-ipfs.

## Changes

This PR implements `Stage 1: Parse CIDs as peer ID` from https://github.com/libp2p/specs/issues/216. It makes `/ipns/{cidv1}` work without waiting for `PeerId.createFromCID` from  https://github.com/libp2p/js-peer-id/pull/105, which means we can merge this before async refactor is done. 

**This PR does not change the default PeerID representation**, 
all it does is to allow use of CIDs in `/ipns/` paths and add tests.

cc https://github.com/ipfs/ipfs/issues/337, https://github.com/libp2p/specs/issues/216, https://github.com/libp2p/specs/pull/209, https://github.com/ipfs/go-ipfs/issues/5287

### CODE REVIEW

**Comment by lidel** [comment]
@jacobheun @alanshaw IIUC  this requires switch to async version of `peer-id`, which basically means bubbling up changes to the rest of libp2p. 

I don't think we want to invest time in backporting this to non-async version, so perhaps I could help with async refactor? I don't want to step on anyone's toes, let me know how to proceed.

**Comment by lidel** [comment]
@alanshaw this PR does not have any external dependencies, 
do you think it could included in [v0.40.0](https://github.com/ipfs/js-ipfs/issues/2558)?

File: test/core/name.spec.js [hunk 1]
```diff
@@ -372,6 +373,16 @@ describe('name', function () {
 
       expect(value).to.exist()
     })
+
+    it('should resolve an ipns path with PeerID as CIDv1 in Base32 correctly', async function () {
+      const res = await node.add(fixture)
+      await node.name.publish(`/ipfs/${res[0].hash}`)
+      let peerCid = new CID(nodeId)
+      if (peerCid.version === 0) peerCid = peerCid.toV1() // future-proofing
+      const value = await ipnsPath.resolvePath(node, `/ipns/${peerCid.toString('base32')}`)
```
[the remark this answers is not in the document]
Comment (lidel) [reply test/core/name.spec.js]: Those are unit tests for `/src/core/ipns/path`, specific to js-ipfs.
`name.resolve` tests are added in https://github.com/ipfs/interface-js-ipfs-core/pull/553

**Comment by lidel** [comment]
@alanshaw i think those are unit tests of js-ipfs code and should stay
([name.spec.js#L359-L364](https://github.com/ipfs/js-ipfs/blob/bc1a355e37354e277f5fa58261236668179601b4/test/core/name.spec.js#L359-L364) tests if resolve of `resolvePath(/ipfs/..)` is idempotent).

I've added `name.resolve` test in https://github.com/ipfs/interface-js-ipfs-core/pull/553.

I will PR ipfs-http-client to unpack multihash from CIDs, so it works with old go-ipfs and pass above tests. (will post link here later)

**Comment by alanshaw** [comment]
@lidel I've just updated the `interface-ipfs-core` dep (to github branch) to check the tests pass. Will release a new version and update again if all is good!

**Pull request merged**